dissect/target/tools/diff.py

```diff
@@ -0,0 +1,961 @@
+) -> Iterator[Record]:
+    """Given a list of targets, compare targets against one another and yield File[Created|Modified|Deleted]Records
+    indicating the differences between them."""
+    if len(targets) < 2:
+        raise ValueError("Provide two or more targets to differentiate between.")
+
+    for target_pair in make_target_pairs(targets, absolute):
+        # Unpack the tuple and initialize the comparison class
+        src_target, dst_target = target_pair
+        comparison = TargetComparison(src_target, dst_target, deep, limit)
+
+        paths = ["/"] if include is None else include
+
+        for path in paths:
+            for directory_diff in comparison.walkdir(path, exclude=exclude):
+                for creation_entry in directory_diff.created:
+                    yield FileCreatedRecord(
+                        path=creation_entry.path,
+                        src_target=src_target.path,
+                        dst_target=dst_target.path,
+                    )
+
+                for deletion_entry in directory_diff.deleted:
+                    yield FileDeletedRecord(
+                        path=deletion_entry.path,
+                        src_target=src_target.path,
+                        dst_target=dst_target.path,
+                    )
+
+                for entry_difference in directory_diff.modified:
+                    yield FileModifiedRecord(
+                        path=entry_difference.path,
+                        diff=entry_difference.diff,
+                        src_target=src_target.path,
+                        dst_target=dst_target.path,
+                    )
+
+
+def differentiate_target_plugin_outputs(
+    *targets: tuple[Target], absolute: bool = False, only_changed: bool = False, plugin: str, plugin_args: str = ""
+) -> Iterator[Record]:
+    """Given a list of targets, yielding records indicating which records from this plugin are new, unmodified or
+    deleted."""
+    for target_pair in make_target_pairs(targets, absolute):
+        src_target, dst_target = target_pair
+        comparison = TargetComparison(src_target, dst_target)
+        yield from comparison.differentiate_plugin_outputs(plugin, plugin_args, only_changed)
+
+
+@catch_sigpipe
+def main() -> None:
+    help_formatter = argparse.ArgumentDefaultsHelpFormatter
+    parser = argparse.ArgumentParser(
+        description="target-diff",
+        fromfile_prefix_chars="@",
+        formatter_class=help_formatter,
+    )
+
+    parser.add_argument(
+        "-d",
+        "--deep",
+        action="store_true",
+        help="Compare file contents even if metadata suggests they have been left unchanged",
+    )
+    parser.add_argument(
+        "-l",
+        "--limit",
+        default=FILE_LIMIT,
+        type=int,
+        help="How many bytes to compare before assuming a file is left unchanged (0 for no limit)",
+    )
+    subparsers = parser.add_subparsers(help="Mode for differentiating targets", dest="mode")
```

Comment:
might be an interesting idea to split the subparsers into multiple different functions for readability...